docs/contracts.md

```diff
@@ -139,17 +138,40 @@ bool check_password(const char *attempt, const char *pswd) {
 
 > If the attacker can measure how long the function takes to reject a guess, they can infer the password one character at a time. This leakage surfaces as a violation of the noninterference property with respect to timing observations.
 
-> A counterexample to Definition 2 could be as follows: Let's say we have two inputs with the same secret value but different public values:
+> A counterexample to Definition 2 could be as follows: Let's say we use the same input on two different secrets:
 
-> - `input1={attempt="abc", pswd="aaa"}`
-> - `input2={attempt="aab", pswd="aaa"}`
+> - `input1={attempt="aaa", pswd="abc"}`
+> - `input2={attempt="aaa", pswd="aab"}`
 
 > The traces of these inputs will be:
 
 > - `trace1 = Trace(check_password, input1) = 1`
 > - `trace2 = Trace(check_password, input2) = 2`
 
-> These inputs constitute a violation of Definition 2, as `trace1 != trace2` even though the two inputs have the same secret value.
+> These inputs constitute a violation of Definition 2, as `trace1 != trace2` even though the two inputs have the same public inputs.
+
+---
+
+> **Example 5B (Timing side channel - Password lenght)**
+
+> Noninterference is able to model different kinds of secret-dependent leaks. Let's take for example a patched version of the previous program:
+
+```c
+bool check_password(const char *attempt, const char *pswd) {
```

Comment:
TBH, it would be better to replace it with a completely different example. I'll try to come up with one later, and let's keep this version for now

Reply:
I agree! we can discuss